src/main/command_executor.cpp

```diff
@@ -130,42 +132,46 @@ bool CommandExecutor::doCommandSingleton(Command* cmd)
 
   // dump the model/proof/unsat core if option is set
   if (status) {
-    Command* g = NULL;
+    std::vector<std::unique_ptr<Command> > getCommands;
     if (d_options.getProduceModels() && d_options.getDumpModels() &&
         (res.asSatisfiabilityResult() == Result::SAT ||
          (res.isUnknown() && res.whyUnknown() == Result::INCOMPLETE))) {
-      g = new GetModelCommand();
+      getCommands.emplace_back(new GetModelCommand());
     }
     if (d_options.getProof() && d_options.getDumpProofs() &&
         res.asSatisfiabilityResult() == Result::UNSAT) {
-      g = new GetProofCommand();
+      getCommands.emplace_back(new GetProofCommand());
     }
 
     if (d_options.getDumpInstantiations() &&
         ((d_options.getInstFormatMode() != INST_FORMAT_MODE_SZS &&
           (res.asSatisfiabilityResult() == Result::SAT ||
            (res.isUnknown() && res.whyUnknown() == Result::INCOMPLETE))) ||
          res.asSatisfiabilityResult() == Result::UNSAT)) {
-      g = new GetInstantiationsCommand();
+      getCommands.emplace_back(new GetInstantiationsCommand());
     }
 
     if (d_options.getDumpSynth() &&
         res.asSatisfiabilityResult() == Result::UNSAT) {
-      g = new GetSynthSolutionCommand();
+      getCommands.emplace_back(new GetSynthSolutionCommand());
     }
 
     if (d_options.getDumpUnsatCores() &&
         res.asSatisfiabilityResult() == Result::UNSAT) {
-      g = new GetUnsatCoreCommand();
+      getCommands.emplace_back(new GetUnsatCoreCommand());
     }
 
-    if (g != NULL) {
+    if (getCommands.size() != 0) {
       // set no time limit during dumping if applicable
       if (d_options.getForceNoLimitCpuWhileDump()) {
         setNoLimitCPU();
       }
-      status = doCommandSingleton(g);
-      delete g;
+      for (const auto& getCommand : getCommands) {
+        status = doCommandSingleton(getCommand.get());
+        if (!status && !d_options.getContinuedExecution()) {
+          break;
```

Comment:
Seems like there should be a warning here if a command has not been executed.

Reply:
I don't think so - if none of the relevant options are set, then no command will be executed.

Reply:
To clarify, what I meant is that if getCommand is not the last command in getCommands and the break happens, maybe a warning is appropriate.

Reply:
Just returning `false` in that case seems consistent with e.g. how we treat command sequences: https://github.com/CVC4/CVC4/blob/480df3b40de9650387c5fceb9aee39311753e3ab/src/main/command_executor.cpp#L75-L81

Reply:
OK, SGTM.